network-api/networkapi/wagtailpages/donation_modal.py

```diff
@@ -35,6 +38,25 @@ class DonationModal(TranslatableMixin, models.Model):
         default="Yes, I'll chip in",
     )
 
+    donate_url = models.CharField(
+        max_length=255,
+        blank=True,
+        validators=[
+            RegexValidator(
+                regex=url_or_query_regex,
+                message=(
+                    "Please enter a valid URL (starting with http:// or https://), "
+                    "or a valid query string starting with ? (Ex: ?form=donate)."
+                ),
```

Comment:
This is the red validation message that will render only if the user enters a invalid value.